tests/test-cluster.py

```diff
@@ -558,28 +573,63 @@ def test_no_cert_reissue_in_nodes(self):
             lock_files = vm.run("ls /var/snap/microk8s/current/var/lock/")
             assert "no-cert-reissue" in lock_files.decode()
 
+    @pytest.mark.skipif(
+        # If the host system does not have IPv6 support or is not configured for IPv6, it won't be able to create VMs with IPv6 connectivity.
+        backend != "lxc" or not is_ipv6_configured,
+        reason="Skipping dual stack tests on VMs which are not lxc based and not dual-stack enabled",
+    )
+    def test_dual_stack_cluster(self):
+        vm = self.VM[0]
+        # Deploy the test deployment and service
+        manifest = TEMPLATES / "dual-stack.yaml"
+        cmd = "lxc file push {} {}/dual-stack.yaml".format(manifest, vm.vm_name).split()
+        subprocess.check_output(cmd)
+        vm.run("/snap/bin/microk8s.kubectl apply -f /dual-stack.yaml")
+
+        # Wait for the deployment to become ready
+        print("Waiting for nginx deployment")
+        while True:
+            ready_pods = 0
+            pods = vm.run("/snap/bin/microk8s.kubectl get po -o wide")
+            for line in pods.decode().splitlines():
+                if "nginxdualstack" in line and "Running" in line:
+                    ready_pods = 1
+            if ready_pods == 1:
+                print(pods.decode())
+                break
+            time.sleep(5)
+
+        # ping the service attached with the deployment
+        ep = (
+            "/snap/bin/microk8s.kubectl get service nginx6 "
+            "-o jsonpath='{.spec.clusterIP}' --output='jsonpath=['{.spec.clusterIP}']'"
+        )
+        ipv6_endpoint = vm.run(ep).decode()
+        print("Pinging endpoint: http://{}/".format(ipv6_endpoint))
+        url = f"http://{ipv6_endpoint}/"
+        attempt = 10
+        while attempt >= 0:
+            try:
+                resp = vm.run("curl {}".format(url))
+                if "Kubernetes IPv6 nginx" in resp.decode():
+                    print(resp)
+                    break
+            except subprocess.CalledProcessError as e:
+                print("Error occurred during the request:", str(e))
+                raise
+            attempt -= 1
+            time.sleep(2)
+
 
 class TestDualStack(object):
```

Comment:
Do we need this TestDualStack anymore? The test was moved to TestCluster so why do keep this class?

Reply:
We can drop the single node tests if we have multinode ones.